### PR TITLE
Added ipresolve option

### PR DESCRIPTION
**Pull request by ggtakec**
### Relevant Issue (if applicable)
#2424 

### Details
Added new option `ipresolve`.
This is an option to fix the IP address version of resolve, and is the `CURLOPT_IPRESOLVE` provided by libcurl.

The default value of the ipresolve option is `whatever`(or `default`), which is the same as `CURL_IPRESOLVE_WHATEVER`.
If `IPv4` (or `v4`) is specified, resolve will be executed using IPv4 only. (`CURL_IPRESOLVE_V4`)
Also, if `IPv6` (or `v6`) is specified, resolve will be executed using IPv6 only. (`CURL_IPRESOLVE_V6`)

### CODE REVIEW

File: src/curl.cpp [hunk 1]
```diff
@@ -1949,7 +1967,11 @@ bool S3fsCurl::ResetHandle(AutoLock::Type locktype)
     if(CURLE_OK != curl_easy_setopt(hCurl, S3FS_CURLOPT_KEEP_SENDING_ON_ERROR, 1) && !run_once){
         S3FS_PRN_WARN("The S3FS_CURLOPT_KEEP_SENDING_ON_ERROR option could not be set. For maximize performance you need to enable this option and you should use libcurl 7.51.0 or later.");
     }
-
+    if(CURL_IPRESOLVE_WHATEVER != S3fsCurl::ipresolve_type){    // CURL_IPRESOLVE_WHATEVER is default, so not need to set.
+        if(CURLE_OK != curl_easy_setopt(hCurl, CURLOPT_IPRESOLVE, S3fsCurl::ipresolve_type)){
```
Comment (gaul) [inline src/curl.cpp:1971]: Should this log when using the non-default value?

File: src/curl.cpp [hunk 2]
```diff
@@ -1174,6 +1175,23 @@ bool S3fsCurl::SetProxyUserPwd(const char* file)
...
+bool S3fsCurl::SetIPResolveType(const char* value)
+{
+    if(!value){
+        return false;
+    }
+    if(0 == strcasecmp(value, "ipv4") || 0 == strcasecmp(value, "v4")){
+        S3fsCurl::ipresolve_type = CURL_IPRESOLVE_V4;
+    }else if(0 == strcasecmp(value, "ipv6") || 0 == strcasecmp(value, "v6")){
+        S3fsCurl::ipresolve_type = CURL_IPRESOLVE_V6;
+    }else if(0 == strcasecmp(value, "whatever") || 0 == strcasecmp(value, "both") || 0 == strcasecmp(value, "default")){
```
Comment (gaul) [inline src/curl.cpp:1187]: Remove unneeded "both" value?  Also do we need `ipv4` and `ipv6`?  This will read weirdly: `ipresolve=ipv4`.

**Review by gaul — CHANGES_REQUESTED** [review]

File: src/curl.cpp (hunk 2 above)
Comment (ggtakec) [reply src/curl.cpp:1187]: This was an option value that I added casually, but it seems inappropriate.
Make it only ipv4/ipv6/whatever.

Comment (ggtakec) [reply src/curl.cpp]: Changed option values, please re-check it.

**Pull request merged**